src/main/java/it/pagopa/pn/deliverypush/actions/LegalFactPdfFromHtmlGeneratorUtils.java

```diff
@@ -0,0 +1,392 @@
+			+ " la PagoPA s.p.a. nella sua qualità di gestore ex lege"
+			+ " ella Piattaforma Notifiche Digitali di cui allo stesso art. 26,"
+			+ " con ogni valore legale per l'opponibilità a terzi, ATTESTA CHE:";
+	
+	private static final String HTML_TEMPLATE = "<html>"
+			+ "<head>"
+			+ "<style type=\"text/css\" media=\"all\">"
+			+ ".container {"
+			+ "width: 635px;"
+			+ "margin: 0 auto;"
+			+ "margin-top: 25px;"
+			+ "}"
+			+ ".paragraph {"
+			+ "font-family: \"Times New Roman\", Times, serif;"
+			+ "font-size: 16px;"
+			+ "margin-top: 25px;"
+			+ "}"
+			+ ".recipient {"
+			+ "margin-top: 25px;"
+			+ "page-break-inside: avoid !important;"
+			+ "display: table;"
+			+ "}"
+			+ ".recipient_row {"
+			+ "display: table-row;"
+			+ "}"
+			+ ".recipient_row > .title {"
+			+ "display: table-cell;"
+			+ "padding: 2pt;"
+			+ "font-weight: bold;"
+			+ "}"
+			+ ".recipient_row > .value {"
+			+ "display: table-cell;"
+			+ "padding: 2pt;"
+			+ "}"
+			+ "p {"
+			+ "margin: 1px;"
+			+ "padding: 1px;"
+			+ "}"
+			+ "hr {"
+			+ "border: 0;"
+			+ "height: 1px;"
+			+ "background: #888888;"
+			+ "margin-bottom: 12px;"
+			+ "}"
+			+ ".footer {"
+			+ "margin-top: 25px;"
+			+ "margin-bottom: 25px;"
+			+ "font-family: \"Times New Roman\", Times, serif;"
+			+ "color: #888888;"
+			+ "font-size: 12px;"
+			+ "}"
+			+ ".footer img {"
+			+ "float: right;    "
+			+ "margin: 0 0 0 15px;"
+			+ "}"
+			+ "#header {"
+			+ "position: running(header);"
+			+ "}"
+			+ "#footer {"
+			+ "margin-top: 25px;"
+			+ "position: running(footer);"
+			+ "}"
+			+ "@page {"
+			+ "size: A4;"
+			+ "margin: 10%;"
+			+ "margin-bottom: 150px;"
+			+ "@top-left {"
+			+ "content: element(header);"
+			+ "}"
+			+ "@bottom-left {"
+			+ "content: element(footer);"
+			+ "}"
+			+ "}"
+			+ "</style>"
+			+ "</head>"
+			+ "<body>"
+			+ "<div class=\"container\">"
+			+ "<div id=\"header\">"
+			+ "<img src=\"pn-logo-header.png\" width=\"180\" height=\"55\" />"
+			+ "</div>"
+			+ "<div class=\"footer\" id=\"footer\">"
+			+ "<hr/>"
+			+ "<img id=\"footer_img\" src=\"pn-logo-footer.png\" width=\"70\" height=\"70\" />"
+			+ "PagoPA S.p.A.<br/>"
+			+ "società per azioni con socio unico<br/>"
+			+ "capitale sociale di euro 1,000,000 interamente versato<br/>"
+			+ "sede legale in Roma, Piazza Colonna 370, CAP 00187<br/>"
+			+ "n. di iscrizione a Registro Imprese di Roma, CF e P.IVA 15376371009"
+			+ "</div>";
+	    
+	private static final String DIV_PARAGRAPH = "<div class=\"paragraph\">";
+	private static final String P_RECIPIENT_ROW = "<p class=\"recipient_row\">";
+	private static final String SPAN_CLASS_VALUE = "<span class=\"value\">%s</span>";
+			
+	@Autowired
+	public LegalFactPdfFromHtmlGeneratorUtils( TimelineDao timelineDao ) {
+        this.timelineDao = timelineDao;
+    }
+    
+	public byte[] generateNotificationReceivedLegalFact(Action action, Notification notification) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 46 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 161-171 into a separate method.